src/data_file.rs

```diff
@@ -24,22 +24,31 @@ pub struct DataFileEnvelope {
 }
 
 #[derive(Debug, Clone)]
-pub struct DataFile {
+pub struct DataFile<MDT>
+where
+  for<'de2> MDT:
+    Deserialize<'de2> + Serialize + PartialEq + Clone + Mergeable + Sized + Send + Sync + 'static,
+{
   pub envelope: DataFileEnvelope,
   pub file: Arc<Mutex<BufReader<File>>>,
   pub data_offset: u64,
+  _phantom: Option<MDT>,
```

Comment:
Is this a convention in Rust?  If not, can we comment about Rust and phantom types?

Reply:
Added comment and the correct Rust pattern: `PhantomData`